protocol/authenticator.go

```diff
@@ -217,13 +217,13 @@ func ResidentKeyUnrequired() *bool {
 
 // Verify on AuthenticatorData handles Steps 9 through 12 for Registration
 // and Steps 11 through 14 for Assertion.
-func (a *AuthenticatorData) Verify(rpIdHash []byte, userVerificationRequired bool) error {
+func (a *AuthenticatorData) Verify(rpIdHash, appIDHash []byte, userVerificationRequired bool) error {
 
 	// Registration Step 9 & Assertion Step 11
 	// Verify that the RP ID hash in authData is indeed the SHA-256
 	// hash of the RP ID expected by the RP.
-	if !bytes.Equal(a.RPIDHash[:], rpIdHash) {
-		return ErrVerification.WithInfo(fmt.Sprintf("RP Hash mismatch. Expected %+s and Received %+s\n", a.RPIDHash, rpIdHash))
+	if !bytes.Equal(a.RPIDHash[:], rpIdHash) && !bytes.Equal(a.RPIDHash[:], appIDHash) {
+		return ErrVerification.WithInfo(fmt.Sprintf("RP Hash mismatch. Expected %s and Received %s\n", a.RPIDHash, rpIdHash))
```

Comment:
This should probably be `return ErrVerification.WithInfo(fmt.Sprintf("RP Hash mismatch. Expected %v but Received %v and %v\n", a.RPIDHash, rpIdHash, appIDHash))` to log the U2F app ID hash and use `%v` to prevent putting binary data in an error message.

Reply:
Good spot, though maybe %x  would be better since %v will just print the uint8 values?

Reply:
Did you have a preference on this? @zachhuff386